src/api/app/models/obs_factory/distribution_strategy_factory.rb

```diff
@@ -113,13 +111,10 @@ def totest_version
     #
     # @return [String] version string
     def published_version
-      begin
-        f = open(repo_url)
-      rescue ::OpenURI::HTTPError
-        return 'unknown'
-      end
-      matchdata = /openSUSE-(.*)-#{published_arch}-.*/.match(f.read)
-      matchdata[1]
+      response = Net::HTTP.get_response(URI.parse(repo_url))
+      return 'unknown' if response.code != '200'
```

Comment:
How confident are you this handles redirects?

Reply:
A suggestion (untested) how to handle it:

```
 require 'net/http'
require 'uri'

def fetch(uri_str, limit = 10)
  # You should choose better exception.
  raise ArgumentError, 'HTTP redirect too deep' if limit == 0

  url = URI.parse(uri_str)
  req = Net::HTTP::Get.new(url.path, { 'User-Agent' => 'Mozilla/5.0 (etc...)' })
  response = Net::HTTP.start(url.host, url.port) { |http| http.request(req) }
  case response
  when Net::HTTPSuccess     then response
  when Net::HTTPRedirection then fetch(response['location'], limit - 1)
  else
    response.error!
  end
end
```
source:  https://stackoverflow.com/questions/6934185/ruby-net-http-following-redirects